docs/usage/csi_components.md

```diff
@@ -0,0 +1,22 @@
+# (Custom) CSI Components
+
+Some provider extensions for Gardener are using CSI components to manage persistent volumes in the shoot clusters.
+Additionally, most of the provider extensions are even deploying controllers for taking volume snapshots (CSI snapshotter).
+
+End-users can deploy their own CSI components and controllers into shoot clusters.
+In such situations, there are multiple controllers acting on the `VolumeSnapshot` custom resources (each responsible for those instances associated with their respective driver provisioner types).
+
+## Recommendations
+
+Custom CSI components are typically regular `Deployment`s running in the shoot clusters.
+
+**Please label them with the `shoot.gardener.cloud/no-cleanup=true` label.**
+
+## Background Information
+
+When a shoot cluster is deleted, Gardener deletes most Kubernetes (`Deployment`s, `DaemonSet`s, `StatefulSet`s, etc.), i.e., CSI components without above mentioned label will be deleted immediately.
+
+If the CSI components (and all resources associated with them) managed by end-users aren't properly removed by the end-users before shoot deletion is triggered then the cluster deletion might cause problems and get stuck.
+
+Typically, it results in stuck `VolumeSnapshot` resources (still having finalizers that will never be cleaned up).
+Consequently, manual intervention is required to clean them up before the cluster deletion can continue.
```

Comment:
Just to clarify: This is the problem you're trying to avoid with the above suggestion, correct? My understanding is that the above label moves the task to deal with the lifecycle of custom CSI components to the end-user. If that's the case, can we more explicitly state that this is what's happening? To me, the logical flow seems to be
* gardener cannot manage custom CSI components for you (because of the issue you described in 'background information')
* therefore, you need to (and probably even want to?) manage these components yourself
* here's how you do it: with a special annotation

Does this make sense?

Reply:
Thanks, WDYT now? I kept the `Background Information` block with the additional information to not overwhelm end-users with too much information upfront. However, I added another sentence motivating why we need such recommendations in the first place.

Reply:
I think what I'd like to see is spelling it out for people that adding this label means: gardener ignores this, you can (and have to) manage the lifecycle on your own for these components. This information might already be in a different place (generically documented for this label), then we can probably just add a link?

Reply:
Can you please make a concrete suggestion (maybe via GitHub) for what you have in mind? :)